src/deployment/deploy.py

```diff
@@ -328,11 +328,11 @@ def setup_rbac(self) -> None:
 
         (password_id, password) = self.create_password(app.object_id)
 
-        cli_app = client.applications.list(filter="appId eq '%s'" % ONEFUZZ_CLI_APP)
-
-        onefuzz_cli_app_uuid = uuid.UUID(ONEFUZZ_CLI_APP)
+        cli_app = list(
+            client.applications.list(filter="appId eq '%s'" % ONEFUZZ_CLI_APP)
+        )
 
-        if not cli_app:
+        if len(cli_app) == 0:
```

Comment:
Can you describe what the bug was? Is is that `client.applications.list()` does not return a `list`, but some kind of generator or other iterable?

Reply:
almost every azure-sdk API returns iterators, not lists.

Reply:
Makes sense. So the `list()` call above is the only thing that _needed_ to be in this PR. I'm fine with the more explicit `len()` check here, it is just equivalent to `not cli_app` for a `list`, so I was wondering if there was anything else going on.

Reply:
yes it was returning an empty iterator  when the no application was found.
the previous test was always evaluating to false